app/src/modules/transport/transport.c

```diff
@@ -124,108 +248,288 @@ static void connect_work_fn(struct k_work *work)
 			  K_SECONDS(CONFIG_APP_TRANSPORT_RECONNECTION_TIMEOUT_SECONDS));
 }
 
-/* Zephyr State Machine framework handlers */
+static void connect_work_cancel(void)
+{
+	k_work_cancel_delayable(&connect_work);
+}
+
+/* Zephyr State Machine Framework handlers */
 
-/* Function executed when the module enters the disconnected state. */
-static void disconnected_entry(void *o)
+/* Handler for STATE_RUNNING */
+
+static void state_running_entry(void *o)
 {
-	struct s_object *user_object = o;
 	int err;
 
-	enum cloud_status status = CLOUD_DISCONNECTED;
-	err = zbus_chan_pub(&CLOUD_CHAN, &status, K_NO_WAIT);
+	ARG_UNUSED(o);
+
+	LOG_DBG("%s", __func__);
```

Comment:
I have a vague memory of this only printing the signature of the parent function, is this still the case?

Reply:
This will print the name of the current function. So this functional will only be compiled in if debug logging is enabled, and serves as a way to debug the state machine